### PR TITLE
ALTAPPS-1006: iOS integrate AppsFlyer

### PR DESCRIPTION
**Pull request by ivan-magda**
**YouTrack Issues**:
[#ALTAPPS-1006](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-1006)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] New analytics events are documented;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**

1. Intergarates AppsFlyer SDK into iOS app
2. Adds `AnalyticEngine` interface to provide platform-specific analytic engines into shared analytic code
3. Logs AppsFlyer events

### CODE REVIEW

**Comment by ivan-magda** [comment]
@XanderZhu ping

File: shared/src/commonMain/kotlin/org/hyperskill/app/analytic/domain/interactor/AnalyticInteractor.kt [hunk 1]
```diff
@@ -43,58 +27,34 @@ class AnalyticInteractor(
...
 
     suspend fun logEvent(event: AnalyticEvent, forceLogEvent: Boolean = false) {
         when (event.source) {
-            AnalyticSource.HYPERSKILL_API -> logHyperskillEvent(event, forceLogEvent)
+            AnalyticSource.HYPERSKILL_API -> {
+                hyperskillAnalyticEngine.reportEvent(event, force = forceLogEvent)
+            }
+            AnalyticSource.APPS_FLYER -> {
+                appsFlyerAnalyticEngine?.reportEvent(event, force = forceLogEvent)
+            }
```
Comment (XanderZhu) [inline shared/src/commonMain/kotlin/org/hyperskill/app/analytic/domain/interactor/AnalyticInteractor.kt:43]: I suggest replace `when` expression with  something like this
```kotlin
analyticEngines
            .filter { it.targetSource == event.source }
            .forEach { it.reportEvent(event, forceLogEvent) }
```
In my opinion, it's better to not expose the engines internals to the Analytics interactor.

File: shared/src/commonMain/kotlin/org/hyperskill/app/analytic/domain/interactor/AnalyticInteractor.kt [hunk 2]
```diff
@@ -43,58 +27,34 @@ class AnalyticInteractor(
...
-
-            val isAuthorized = authInteractor.isAuthorized()
-                .getOrDefault(false)
-
-            hyperskillRepository.flushEvents(isAuthorized)
+    override fun setAppTrackingTransparencyAuthorizationStatus(isAuthorized: Boolean) {
+        if (hyperskillAnalyticEngine is HyperskillAnalyticEngine) {
+            hyperskillAnalyticEngine.setAppTrackingTransparencyAuthorizationStatus(isAuthorized = isAuthorized)
         }
     }
```
Comment (XanderZhu) [inline shared/src/commonMain/kotlin/org/hyperskill/app/analytic/domain/interactor/AnalyticInteractor.kt:59]: In my opinion, it's better to not expose the internals of the engine to the Analytics interaction.
We can add `setAppTrackingTransparencyAuthorizationStatus` to the `AnalyticEngine` interface. 
And only Hyperskill implementation will do some work.
And I think, we can do the same with `setScreenOrientation `.

**Review by XanderZhu — COMMENTED** [review]
Looks good!

**Review by XanderZhu — APPROVED** [review]

**Pull request merged**